main.py

```diff
@@ -91,72 +96,75 @@ def _get_link_from_interface(self, interface):
                 return link
         return None
 
-    def _restore_links(self):
-        """Restore link saved in StoreHouse."""
-        for link_id, state, in self.links_state.items():
-            dpid_a = state['endpoint_a']['switch']
-            iface_id_a = int(state['endpoint_a']['id'][-1])
-            dpid_b = state['endpoint_b']['switch']
-            iface_id_b = int(state['endpoint_b']['id'][-1])
-            try:
-                endpoint_a = self.controller.switches[dpid_a].interfaces[
-                    iface_id_a]
-                endpoint_b = self.controller.switches[dpid_b].interfaces[
-                    iface_id_b]
-            except KeyError as error:
-                error_msg = (f"Error restoring link endpoint: {error}")
-                raise KeyError(error_msg)
-
-            link = self._get_link_or_create(endpoint_a, endpoint_b)
-            endpoint_a.update_link(link)
-            endpoint_b.update_link(link)
-
-            endpoint_a.nni = True
-            endpoint_b.nni = True
-
-            self.notify_topology_update()
+    def _restore_links(self, link_id):
```

Comment:
As this method now restores just one link.
```suggestion
    def _restore_link(self, link_id):
```